### PR TITLE
Accuchecker API

### PR DESCRIPTION
**Pull request by hugeox**
Updates blacklist cache and elastic with report sent to /blacklist/accuracyupdate/ from Accuchecker.
In terms of data validation this is taken care of by gson and using a new class IoCAccuCheckerReport which mimics the IoCRecord class but only allows fields source,accuracy and metadata. 

+ fix pushiocs.sh space bug

### CODE REVIEW

File: backp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                update.setUpsert(upsert);
+            }
+            update.execute().actionGet();
+            return true;
+        } catch (Exception ex) {
+            log.log(Level.SEVERE, "Upsert went wrong: " + ex.getMessage());
+            ex.printStackTrace();
+            throw new ArchiveException("Upsert went wrong: " + ex.getMessage());
+        }
+    }
```
Comment (Karm) [inline backp]: @hugeox This file was added by mistake apparently.

File: ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java [hunk 2]
```diff
@@ -148,6 +154,14 @@ public EventLogRecord archiveEventLogRecordUsingLogstash(EventLogRecord logRecor
         return logRecord;
     }
 
+    @Override
+    public List<IoCRecord> getMatchingEntries(String name, String value) throws ArchiveException {
+        // lists iocs matching query
+        //assume less than 1000 results
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java]: @hugeox Could you elaborate more? 1000 results or what?

File: ejb/src/main/java/biz/karms/sinkit/ejb/impl/CoreServiceEJB.java [hunk 3]
```diff
@@ -145,6 +146,37 @@ public IoCRecord processIoCRecord(final IoCRecord ioc) throws ArchiveException,
         return ioc;
     }
 
+    @Override
+    public boolean updateWithAccuCheckerReport(IoCAccuCheckerReport report) throws ArchiveException, IoCValidationException {
+        //returns true if all archivations and saves to cache succeed, otherwise returns false(or throws ArchiveException)
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/ejb/impl/CoreServiceEJB.java]: @hugeox Move to JavaDoc, please.

File: ejb/src/main/java/biz/karms/sinkit/ejb/impl/CoreServiceEJB.java [hunk 4]
```diff
@@ -145,6 +146,37 @@ public IoCRecord processIoCRecord(final IoCRecord ioc) throws ArchiveException,
...
+        if (report.getSource() == null || report.getSource().getId() == null || report.getSource().getId().getValue() == null) {
+            throw new IoCValidationException("Accuchecker report doesn't have required field source.id.value.");
+        }
+        if (report.getAccuracy() == null) {
+            throw new IoCValidationException("Accuchecker report doesn't have required field accuracy.");
+        }
+        boolean response = true;
+        final HashMap<String, Integer> report_accuracy = report.getAccuracy();
+        String source_id_value = report.getSource().getId().getValue();
+        List<IoCRecord> iocs = archiveService.getMatchingEntries("source.id.value", source_id_value);
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/ejb/impl/CoreServiceEJB.java:168]: @hugeox The index is not used, fori would be better replaced with a lambda if you please.

File: ejb/src/main/java/biz/karms/sinkit/ioc/IoCAccuCheckerReport.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package biz.karms.sinkit.ioc;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+public class IoCAccuCheckerReport implements Serializable {
+
+    //same structure as IoCRecord, but only allowing 3 fields.
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/ioc/IoCAccuCheckerReport.java]: @hugeox Move to JavaDoc, please.

File: ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java [hunk 6]
```diff
@@ -105,6 +105,12 @@ public boolean setVirusTotalReportToIoCRecord(final IoCRecord ioc, final IoCViru
         return elasticService.update(ioc.getDocumentId(), vtReports, ELASTIC_IOC_INDEX, ELASTIC_IOC_TYPE, null);
     }
```
Comment (hugeox) [inline ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java]: This should only accept IoCAccuCheckerReport to guarantee that hashes do not need to be recomputed.

File: ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java [hunk 7]
```diff
@@ -105,6 +105,12 @@ public boolean setVirusTotalReportToIoCRecord(final IoCRecord ioc, final IoCViru
...
 
+    @Override
+    public boolean setReportToIoCRecord(final Object report,String document_id) throws ArchiveException{
+        // a wrapper for simple update
+        return elasticService.update(document_id, report, ELASTIC_IOC_INDEX, ELASTIC_IOC_TYPE, null);
+
+    }
```
Comment (hugeox) [inline ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java]: This should only accept IoCAccuCheckerReport to guarantee that hashes do not need to be recomputed.

**Review by Karm — CHANGES_REQUESTED** [review]
@hugeox 
 * attend to the notes from review please, especially to superfluous file etc.
 * also rebase so as the branch doesn't have conflicts
 * reopen against master, please
 * reword commits, e.g. "First steps" :-1:

**Review by Karm — APPROVED** [review]

**Pull request merged**